### PR TITLE
Restore backward compartibility beteen 192 and 193 Rider SDK

### PR DESCRIPTION
**Pull request by sdubov**
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
* Bump rd-gen library version to 193 compartible generated models. Further updates require updating min support version to 193.
* Re-work protocol calls to use event sourcing apptoach instead of protocol calls instances. It was changed to restore compartibility between version in Rider. New protocol version has modifications in "call" instances that are not backward compartible, so the logic was changed to other entities
* Update Rider model generation logic to make one rd-gen gradle task call and to avoid models re-generation
* Re-format completion logic and move it under PSI model

## Motivation and Context
Latest changes in Rider protocol and SDK cause breaking the compatibility between 192 and 193 SDK. This commit restore compatibility by re-writing protocol "call" with an instance that were not affected by breaking changes in protocol.

## Related Issue(s)
None.

## Testing
* Verified compilation for 193 and 192 SDK.
* Verified related backend logic (completion and run configuration)

## Screenshots (if appropriate)
None.

## Checklist
- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sdubov** [comment]
Please note, the compilation will break until new Rider SDK release with latest changes to support backward compatibility. This commit is for review purpose for now.

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolver.kt [hunk 1]
```diff
@@ -4,25 +4,29 @@
...
 import com.intellij.psi.search.GlobalSearchScope
 import com.jetbrains.rd.framework.impl.RpcTimeouts
-import com.jetbrains.rider.model.backendPsiHelperModel
-import com.jetbrains.rider.model.publishableProjectsModel
-import com.jetbrains.rider.model.HandlerExistRequest
-import com.jetbrains.rider.model.MethodExistingRequest
+import com.jetbrains.rd.util.AtomicInteger
+import com.jetbrains.rd.util.spinUntil
+import com.jetbrains.rd.util.threading.SynchronousScheduler
+import com.jetbrains.rider.model.*
```
Comment (abrooksv) [inline jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolver.kt]: We avoid *

File: jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt [hunk 2]
```diff
@@ -3,27 +3,95 @@
...
+object PsiKotlinGenerator : ExternalGenerator(
+    Kotlin11Generator(FlowTransform.AsIs, "com.jetbrains.rider.model", File(syspropertyOrInvalid("ktGeneratedOutput"))),
+    IdeRoot)
+
+object PsiCSharpGenerator : ExternalGenerator(
+    CSharp50Generator(FlowTransform.Reversed, "JetBrains.Rider.Model", File(syspropertyOrInvalid("csPsiGeneratedOutput"))),
+    IdeRoot)
+
 @Suppress("unused")
 object LambdaPsiModel : Ext(SolutionModel.Solution) {
```
Comment (abrooksv) [inline jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt]: should the protocol be a source set so IDE sets up dependencies right?

File: jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt [hunk 3]
```diff
@@ -3,27 +3,95 @@
...
+
     init {
-        call("isHandlerExists", HandlerExistRequest, bool).async
-            .doc("Check whether handler with specified name exists for a partucular project")
+        setting(GeneratorBase.AcceptsGenerator) { generator ->
+            generator == PsiKotlinGenerator.generator ||
+            generator == PsiCSharpGenerator.generator
+        }
+
+        // TODO: This event sourcing model is a trade of for compartibility reasons between 192 and 193 Rider SDK.
```
Comment (abrooksv) [inline jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt]: trade off*

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolver.kt (hunk 1 above)
Comment (sdubov) [reply jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolver.kt]: Fixed.

File: jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt (hunk 3 above)
Comment (sdubov) [reply jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt]: Fixed.

File: jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt (hunk 2 above)
Comment (sdubov) [reply jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt]: I'm sorry, not sure I got the question. Model in this context is used to generate classes that are used on both side from the protocol: Kotlin – protocol – C#.

**Comment by sdubov** [comment]
Verified for both 192 and 193 Rider SDKs.

File: jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt (hunk 2 above)
Comment (sdubov) [reply jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt]: Well... yes, if I correctly understand your question, you might say it that way. The way you define an exact model - set up how the protocol works inside IDE.

Comment (abrooksv) [reply jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt]: How did you set it up in the IDE? Are you editing the protocol in rider or idea? For me the classpath isn't set up so it is missing the kotlin sdk and Rd depenedency

Comment (abrooksv) [reply jetbrains-rider/protocol/model/psi/LambdaPsiModel.kt]: I mean from a Gradle perspective

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/completion/DotNetHandlerCompletion.kt [hunk 4]
```diff
@@ -8,18 +8,48 @@ import com.intellij.codeInsight.completion.impl.CamelHumpMatcher
...
 import com.jetbrains.rdclient.icons.FrontendIconHost
-import com.jetbrains.rider.model.lambdaModel
+import com.jetbrains.rider.model.DetermineHandlersRequest
+import com.jetbrains.rider.model.HandlerCompletionItem
+import com.jetbrains.rider.model.lambdaPsiModel
 import com.jetbrains.rider.projectView.solution
 
 class DotNetHandlerCompletion : HandlerCompletion {
 
+    companion object {
```
Comment (abrooksv) [inline jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/completion/DotNetHandlerCompletion.kt]: Make these private

**Pull request merged**